iidm/iidm-test/src/main/java/com/powsybl/iidm/network/test/ScadaNetworkFactory.java

```diff
@@ -0,0 +1,243 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.test;
+
+import com.powsybl.iidm.network.*;
+import org.joda.time.DateTime;
+
+import java.util.Objects;
+
+/**
+ * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
+ */
+public final class ScadaNetworkFactory {
+
+    public static Network create() {
+        return create(NetworkFactory.findDefault());
+    }
+
+    public static Network create(NetworkFactory networkFactory) {
+        Objects.requireNonNull(networkFactory);
+        Network network = Network.create("scada", "test");
+        network.setCaseDate(DateTime.parse("2017-06-25T17:43:00.000+01:00"));
+        network.setMinimumAcceptableValidationLevel(ValidationLevel.SCADA);
+        Substation sub = network.newSubstation()
+                .setId("sub")
+                .add();
+        VoltageLevel vl = sub.newVoltageLevel()
+                .setNominalV(440.0)
+                .setId("vl")
+                .setTopologyKind(TopologyKind.BUS_BREAKER)
+                .add();
+        String busId = "bus";
+        vl.getBusBreakerView()
+                .newBus()
+                .setId(busId)
+                .add();
+        vl.newBattery()
+                .setId("battery")
+                .setMinP(0.0)
+                .setMaxP(100.0)
+                .setBus(busId)
+                .add();
+        DanglingLine dl = vl.newDanglingLine()
+                .setId("dl")
+                .setBus(busId)
+                .setR(1.0)
+                .setX(1.0)
+                .setG(0.0)
+                .setB(0.0)
+                .newGeneration()
+                .setMinP(100.0)
+                .setMaxP(200.0)
+                .add()
+                .add();
+        dl.getGeneration().newMinMaxReactiveLimits()
+                .setMinQ(100)
+                .setMaxQ(200)
+                .add();
+        Generator generator = vl.newGenerator()
+                .setVoltageRegulatorOn(true)
```

Comment:
Where is the regulating terminal here? It is an equipment information. Same remark for shunt compensator, phase and ratio tap changers, VSC converter station. Without this information, the network is just invalid. It is what you want to show here?

Reply:
Indeed this factory is not used, right?

Reply:
It is used in `AbstractNetworkTest`